handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java

```diff
@@ -34,6 +35,9 @@
  * Triggers an {@link IdleStateEvent} when a {@link Channel} has not performed
  * read, write, or both operation for a while.
  *
+ * The {@link IdleStateHandler} should be placed as first {@link ChannelHandler}
```

Comment:
Depends. :D

I don't think it's bad advise if you keep DoS attack vectors, malicious or buggy behavior out of the picture. But ultimately it boils down to realizing that idleness is measured on event level and the meaning thereof depends on the position in the pipeline.